skbio/stats/tests/test_composition.py

```diff
@@ -269,6 +355,136 @@ def test_centralize(self):
                             np.array([[2, 2, 6],
                                       [4, 4, 2]]))
 
```

Comment:
All of the tests use `grouping` with groups defined in contiguous blocks, e.g.:

```python
self.cats1 = pd.Series([0, 0, 0, 1, 1, 1])
```

Can you add a test to make sure the results are the same when passing the same grouping structure in a non-contiguous fashion, e.g.:

```python
self.cats1 = pd.Series([0, 0, 1, 1, 0, 1])
```

I don't have any reason to believe this won't work with the current implementation but doesn't hurt to test this in case the code inadvertently relies on contiguous groups in the future.